lib/unifex/unifex_cnode.ex

```diff
@@ -0,0 +1,168 @@
+defmodule Unifex.UnifexCNode do
+  @doc """
+  Wraps Bundlex.CNode functionalities, in due to support specific Unifex's CNode behaviours
+  """
+
+  require Bundlex.CNode
+
+  @enforce_keys [:server, :node]
+  defstruct @enforce_keys
+
+  @type t :: %__MODULE__{
+          server: pid,
+          node: node
+        }
+
+  @type on_start_t :: {:ok, t} | {:error, :spawn_cnode | :connect_to_cnode}
+
+  def cast_on_start_t({:ok, %Bundlex.CNode{} = bundex_cnode}) do
+    {:ok, cast_cnode(bundex_cnode)}
+  end
+
+  def cast_on_start_t({:ok, %__MODULE__{} = unifex_cnode}) do
+    {:ok, cast_cnode(unifex_cnode)}
+  end
+
+  def cast_on_start_t(on_start) do
+    on_start
+  end
+
+  def cast_cnode(%Bundlex.CNode{server: server, node: node}) do
+    %__MODULE__{
+      server: server,
+      node: node
+    }
+  end
+
+  def cast_cnode(%__MODULE__{server: server, node: node}) do
+    %Bundlex.CNode{
+      server: server,
+      node: node
+    }
+  end
+
+  @doc """
+  Calls Bundlex.CNode functions.
+  Look at Bundlex.CNode docs, to see more.
+  """
+
+  defmacro start_link(native_name) do
+    quote do
+      require Bundlex.CNode
+
+      unquote(native_name)
+      |> Bundlex.CNode.start_link()
+      |> unquote(__MODULE__).cast_on_start_t
+    end
+  end
+
+  defmacro start(native_name) do
+    quote do
+      require Bundlex.CNode
+
+      unquote(native_name)
+      |> Bundlex.CNode.start()
+      |> unquote(__MODULE__).cast_on_start_t
+    end
+  end
+
+  @spec stop(t) :: :ok | {:error, :disconnect_cnode}
+  def stop(%__MODULE__{} = unifex_cnode) do
+    unifex_cnode
+    |> cast_cnode
+    |> Bundlex.CNode.stop()
+  end
+
+  @spec monitor(t) :: reference
+  def monitor(%__MODULE__{} = unifex_cnode) do
+    unifex_cnode
+    |> cast_cnode
+    |> Bundlex.CNode.monitor()
+  end
+
+  @spec send(t, message :: term) :: :ok
+  def send(%__MODULE__{} = unifex_cnode, message) do
+    unifex_cnode
+    |> cast_cnode
+    |> Bundlex.CNode.send(message)
+  end
+
+  def unpack_result({:result, content}) do
+    content
+  end
+
+  def unpack_message({:send, content}) do
+    content
+  end
+
+  @doc """
+      Sends to CNode message containing data about which function should be called and with which args,
+      then waits timeout miliseconds to receive and unpack function result
```

Comment:
Put more focus on the point (what?) instead of the process (how?). The end-user doesn't need to know that some message is sent and the result is packed anyhow.